broker/lib/middleware.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const _ = require('lodash');
+const errors = require('../../common/errors');
+const logger = require('../../common/logger');
+const quota = require('../lib/quota');
+const quotaManager = quota.quotaManager;
+const CONST = require('../../common/constants');
+const Forbidden = errors.Forbidden;
+const BadRequest = errors.BadRequest;
+const utils = require('../../common/utils');
+const catalog = require('./models/catalog');
```

Comment:
should we move `models` also to common? This is something that could get shared with all?

Reply:
I feel that `model`, `catalog`, `plan` etc. are very `broker` specific. Just to give an example, service definitions will not be a part of deployment hook job `config`

Reply:
same as above.